File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/java/F05.P3.java
import java.util.ArrayList;
import java.util.List;
import java.util.function.IntUnaryOperator;

class Main {
    static int probe() {
        List<Integer> xs = new ArrayList<>(List.of(1, 2, 3));
        int k = 0;
        // BEGIN PROBE F05.P3
        IntUnaryOperator neg = a -> k - a;
        int[] ys = xs.stream().mapToInt(Integer::intValue).map(neg).toArray();
        return ys[0];
        // END PROBE F05.P3
    }

    public static void main(String[] args) {
        System.out.println(probe());
    }
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/kotlin/F05.P3.kt
fun probe(): Int {
    val xs = mutableListOf(1, 2, 3)
    val k = 0
    // BEGIN PROBE F05.P3
    val neg = { a: Int -> k - a }
    val ys = xs.map(neg)
    return ys[0]
    // END PROBE F05.P3
}

fun main() {
    println(probe())
}
